sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2779,7 +2800,8 @@ def _provision(
                     rich_utils.force_update_status(
                         ux_utils.spinner_message('Launching', log_path))
                     config_dict = retry_provisioner.provision_with_retries(
-                        task, to_provision_config, dryrun, stream_logs)
+                        task, to_provision_config, dryrun, stream_logs,
+                        skip_if_config_hash_matches)
```

Comment:
I feel like it complicates the `to_provision_config` too much.
Currently, `to_provision_config` is set from `_check_existing_cluster` and it's not modified anywhere else, which is helpful to reason about. It seems irrelevant to pass `skip_if_no_updates` into `_check_existing_cluster`, and bad to modify `to_provision_config` once it has been created.

We could move this code on L2755-2759 into `provision_with_retries`, and pass `skip_if_no_updates` directly. But this complicates `provision_with_retries`, which currently only deals with cloud-specific retry things.

So my opinion is that this is the cleanest way to do it, but I don't feel that strongly.

Reply:
It seems we do store the `prev_config_hash` in `to_provision_config` at the moment. Should we just pass the `skip_if_no_cluster_updates` to the underlying function, instead of the hash itself?

Reply:
Yeah, this is what I meant by the second paragraph in my comment above. It means that provision_with_retries has to handle more business logic besides just setting up retries and calling provision. But we could do it.

Reply:
I am leaning towards to pushing down the functionality as we are only comparing it in that function, i.e. keeping the function deep.